src/vm/comtoclrcall.cpp

```diff
@@ -679,15 +678,11 @@ extern "C" UINT64 __stdcall COMToCLRWorker(Thread *pThread, ComMethodFrame* pFra
     //
     HRESULT hr = S_OK;
 
-    pThread = GetThread();
-    if (NULL == pThread)
+    pThread = SetupThreadNoThrow();
```

Comment:
I would think we want to keep calling `GetThread()` here first unless you have collected performance numbers that show it is not worth it. `GetThread()` is small inlineable method. SetupThreadNoThrow is non-inlineable, with fat frame due to exception handling.

Reply:
I am not sure I could even measure that impact - either way. There are so many layers that honestly it would wind up being noise more than anything. I am fine putting it back, but what I can say here is that so many of these functions have precisely the same redundant checks and so if we removed them I would surmise the overall code size reduction would be more valuable than possible inlining optimization in an already expensive code path.

Reply:
Somebody was clearly doing work to squeeze cycles out of this function in the past: `COMToCLRWorkerBody` is marked as `FORCEINLINE`, and it calls other carefully factored functions that are marked as `FORCEINLINE` or `NOINLINE`

Reply:
> Somebody was clearly doing work to squeeze cycles out of this function in the past: 

That could be true. However as a counter point I can also point at several places where that is a no-op or even a degenerate operation because of the code bloat generated. Also we have LTCG enabled and the vast majority of these `FORCEINLINE`s will either melt away or just aren't needed because the compiler is smart enough to do the right thing. I also can appreciate the time frame when the MSVC compiler was probably less smart, but there are many cases in this codebase where these kind of optimizations was of dubious value or had non-measurable impact at best.